middleware.go

```diff
@@ -3,10 +3,27 @@ package server
 import (
 	"fmt"
 	"net/http"
+
+	"github.com/juju/errgo"
 )
 
+// NewWelcomeMiddleware provides a middleware that responds human readable
+// information about a service. E.g. one can register this under /.
 func NewWelcomeMiddleware(appName, version string) Middleware {
 	return func(res http.ResponseWriter, rep *http.Request, ctx *Context) error {
 		return ctx.Response.PlainText(fmt.Sprintf("This is %s version %s\n", appName, version), http.StatusOK)
 	}
 }
+
+// NewHealthcheckMiddleware provides a middleware that responds JSON formatted
+// information about a service. E.g. one can register this under /healthcheck.
+func NewHealthcheckMiddleware(hc Healthchecker) Middleware {
+	return func(res http.ResponseWriter, rep *http.Request, ctx *Context) error {
+		hcRes, err := hc.Status()
+		if err != nil {
+			return errgo.Mask(err)
+		}
+
+		return ctx.Response.Json(hcRes, http.StatusOK)
```

Comment:
Do you think it would make sense to respond with a http://httpstatus.es/503 Service Unavailable if the service is down? So one can see from the HTTP Response Code w/o looking at the body?